### DISCUSSION TITLE
run failed: workspace configuration error: package.json: no workspaces found. Turborepo requires npm workspaces to be defined in the root package.json

### DISCUSSION THREAD

**Discussion by zhangjie25** [xyflow/xyflow#2478] [Q&A]
any help on this when start pnpm test
run failed: workspace configuration error: package.json: no workspaces found. Turborepo requires npm workspaces to be defined in the root package.json

**Comment by standard-software** [comment]
I have the same symptom and was able to solve it, so I will let you know.

I added the following to package.json in root and now it works.
```
  "workspaces": ["packages/*"],
```

It seems to work without configuration once it works, so it may be a problem that occurs only right after pnpm is installed.

**Comment by moklick** [comment]
Are you using `pnpm`? The workspaces are defined under `pnpm-workspace.yaml`.

**Comment by mikemaccana** [comment]
@moklick Yes, pnpm uses `pnpm-workspace.yaml` but ... Turborepo requires npm workspaces to be defined in the root package.json

**Reply by bcakmakoglu** [reply]
Where does it say that? I'm using turbo in my repo as well and I did not define npm workspaces and it works perfectly fine 🤔

**Reply by gynekolog** [reply]
Confirm that without the workspaces defined in package.json, some commands won't work.
For example `pnpm version` with `-ws` option.

**Comment by cajotafer** [comment]
Maybe you need to remove `packageManager` from your `package.json`
